django/test/testcases.py

```diff
@@ -138,10 +139,21 @@ def on_template_render(self, sender, signal, template, context, **kwargs):
         self.rendered_templates.append(template)
         self.context.append(copy(context))
 
+    @property
+    def rendered_template_names(self):
+        rendered_names = []
+        for t in self.rendered_templates:
+            if t.name is None:
+                continue
+            if isinstance(t, PartialTemplate):
+                rendered_names.append(f"{t.origin.template_name}#{t.name}")
+            rendered_names.append(t.name)
```

Comment:
Factored this property out once we needed it in two places. Also nice is that this "qualified" form now appears first in the output before the "short" form.

Reply:
I would rather to make this a list comprehension as follow (untested but looks correct?):
```python
rendered_names = [
    f"{t.origin.template_name}#{t.name}" if isinstance(t, PartialTemplate) else t.name
    for t in self.rendered_templates
    if t.name
]
```